docs/src/parameter_distributions.md

```diff
@@ -449,6 +463,51 @@ using Plots
 plot(u, constrained = false)
 ```
 
+## [Function Distribution Example](@id function-example)
+Here, define a function parameter distribution on ``[0,1] \times [1,2]`` , bounded by ``[-5,-3]`` and with correlation lengthscales 0.05. First, we get the packages:
+
+```@example snip_fun
+using EnsembleKalmanProcesses.ParameterDistributions # For `ParameterDistribution`
+using Random, Distributions # for `rand` and `Normal`
+using Plots
+# IMPORT the GRF package
+import GaussianRandomFields # for `GaussianRandomFields`
+```
+then, we use the  [`GaussianRandomFields.jl`](https://github.com/PieterjanRobbe/GaussianRandomFields.jl) package to define the distribution of choice. This distribution is unbounded. Here we take a `Matern` kernel, and define our evaluation grid on the domain. We choose 30 degrees of freedom (`dofs`), so this function distribution is specified through the value of 30 learnable coefficients. 
+```@example snip_fun
+const GRF = GaussianRandomFields
+# Define a `GaussianRandomFields` object
+input_dim = 2 # Define a 2D -> 1D function
+dofs = 30 # the number of modes defining the distribution
+points = [collect(0:0.01:1), collect(1:0.02:2)] # the 2D domain grid (uniform in each dimension)
+
+grfjl_obj = GRF.GaussianRandomField(
+   GRF.CovarianceFunction(input_dim, GRF.Matern(0.05, 2)),
+   GRF.KarhunenLoeve(dofs),
+   points...,
+) # the Gaussian Random Field object from the package
+nothing # hide
+```
+We define our parameter distribution wrapper, where `GRFJL()` indicates the GRF package used. We also impose bounds into an interval ``[-5,-3]`` (here applied to the output space).
+```@example snip_fun
+grf = GaussianRandomFieldInterface(grfjl_obj, GRFJL()) # our wrapper
+pd = ParameterDistribution(
+    Dict(
+        "distribution" => grf,
+        "constraint" => bounded(-5, -3), 
+        "name" => "func_in_min5_min3",
+    )
+) # The ParameterDistribution with constraint in the output space
+nothing # hide
+```
+We plot 4 samples of this distribution. Samples are taken over the (30-dimensional) degrees of freedom, and then we apply the `transform_unconstrained_to_costrained` map to (i) build the function distribution, (ii) evaluate it on the numerical grid, and (iii) constrain the output with our prescribed bounds.
+```@example snip_fun
+shape = [length(pp) for pp in points]
```

Comment:
could `ndims` be used here instead to illustrate the usage of that method?

Reply:
the functional parameter `ndims` only gives the total number of eval points, where as here for plotting we need the breakdown of # eval points per dimension. It might be worth storing this grid-shape internally however, for future use in reshaping. I'll create a separate issue, I presume it is obtainable from the GRF package.

Reply:
I have appended this into #282